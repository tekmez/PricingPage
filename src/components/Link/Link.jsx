import { Linking, Text, TouchableOpacity } from 'react-native'
import styles from './Link.style'
const Link = ({ url, title }) => {
  return (
    <TouchableOpacity testID="link" onPress={() => Linking.openURL(url)}>
      <Text testID="link-title" style={styles.link}>
        {title}
      </Text>
    </TouchableOpacity>
  )
}

export default Link
