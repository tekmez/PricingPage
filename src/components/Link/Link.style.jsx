import { StyleSheet } from 'react-native'

const styles = StyleSheet.create({
  link: {
    fontSize: 12,
    lineHeight: 16,
    color: '#717578',
    fontFamily: 'Inter-Medium',
  },
})

export default styles
