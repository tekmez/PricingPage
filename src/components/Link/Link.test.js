import React from 'react'
import { render, fireEvent } from '@testing-library/react-native'
import Link from './Link'
import { Linking } from 'react-native'

describe('Link component', () => {
  it('should match snapshot', () => {
    const comp = render(<Link />).toJSON()
    expect(comp).toMatchSnapshot()
  })

  it('should render correctly', () => {
    const testText = 'Click me'
    const comp = render(<Link title={testText} />)
    const linkText = comp.getByTestId('link-title').children[0]
    expect(linkText).toBe(testText)
  })

  it('should open url on press', () => {
    const testUrl = 'https://www.google.com'
    jest.spyOn(Linking, 'openURL')
    const comp = render(<Link url={testUrl} />)
    const link = comp.getByTestId('link')
    fireEvent.press(link)
    expect(Linking.openURL).toBeCalledWith(testUrl)
  })
})
