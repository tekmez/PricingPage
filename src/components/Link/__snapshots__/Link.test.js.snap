// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component should match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="link"
>
  <Text
    style={
      {
        "color": "#717578",
        "fontFamily": "Inter-Medium",
        "fontSize": 12,
        "lineHeight": 16,
      }
    }
    testID="link-title"
  />
</View>
`;
